src/Controller/Admin/Asset/AssetController.php

```diff
@@ -1516,6 +1530,31 @@ public function getPreviewDocumentAction(Request $request): StreamedResponse
         }
     }
 
+    private function getResponseByScanStatus(Asset\Document $asset) :?Response
+    {
+        if (!Config::getSystemConfiguration('assets')['document']['scan_pdf']) {
+            return null;
+        }
+
+        $scanStatus = $asset->getCustomSetting(Asset\Document::CUSTOM_SETTING_PDF_SCAN_STATUS);
+        if ($scanStatus === null) {
+            $scanStatus = Asset\Enum\PdfScanStatus::IN_PROGRESS->value;
+            \Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
+                new AssetUpdateTasksMessage($asset->getId())
+            );
+        }
+
+        if ($scanStatus === Asset\Enum\PdfScanStatus::IN_PROGRESS->value) {
+            return $this->render('@PimcoreAdmin/admin/asset/get_preview_pdf_in_progress.html.twig');
+        }
+
+        if ($scanStatus === Asset\Enum\PdfScanStatus::UNSAFE->value) {
+            return $this->render('@PimcoreAdmin/admin/asset/get_preview_pdf_unsafe.html.twig');
+        }
+
+        return null;
```

Comment:
Seems like `PdfScanStatus` is missing from this PR.

Anyway, I wonder if we can write it something like this instead:

```suggestion
        $scanStatus = Asset\Enum\PdfScanStatus::tryFrom($asset->getCustomSetting(Asset\Document::CUSTOM_SETTING_PDF_SCAN_STATUS));
        if ($scanStatus === null) {
            $scanStatus = Asset\Enum\PdfScanStatus::IN_PROGRESS;
            \Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
                new AssetUpdateTasksMessage($asset->getId())
            );
        }

        return match($scanStatus) {
            Asset\Enum\PdfScanStatus::IN_PROGRESS => $this->render('@PimcoreAdmin/admin/asset/get_preview_pdf_in_progress.html.twig'),
            Asset\Enum\PdfScanStatus::UNSAFE => $this->render('@PimcoreAdmin/admin/asset/get_preview_pdf_unsafe.html.twig'),
            default => null,
        };
```

Reply:
`PdfScanStatus` comes from [Core PR](https://github.com/pimcore/pimcore/pull/16053) which is not merged yet. Yes, the suggestion looks good to me :+1: